common/v2/services/RatesProvider.tsx

```diff
@@ -1,30 +1,77 @@
 import React, { createContext, useContext, useState, useEffect } from 'react';
 import { isEmpty } from 'lodash';
 
-import { StoreContext, AccountContext, SettingsContext } from 'v2/services/Store';
+import { StoreContext, SettingsContext } from 'v2/services/Store';
 import { PollingService } from 'v2/workers';
-import { IRates, TTicker } from 'v2/types';
+import { IRates, TTicker, Asset } from 'v2/types';
+import { notUndefined } from 'v2/utils';
+import { checkHttpStatus, parseJSON } from './ApiService/utils';
 
 interface State {
   rates: IRates;
   getRate(ticker: TTicker): number | undefined;
+  getAssetRate(asset: Asset): number | undefined;
+}
+
+interface AssetMappingObject {
+  coinGeckoId?: string;
+  cryptoCompareId?: string;
+  coinCapId?: string;
+}
+
+interface AssetMappingListObject {
+  [key: string]: AssetMappingObject;
 }
 
 const DEFAULT_FIAT_PAIRS = ['USD', 'EUR'] as TTicker[];
 const DEFAULT_FIAT_RATE = 0;
 const POLLING_INTERRVAL = 60000;
-const RATES_URL = 'https://proxy.mycryptoapi.com/cc/multi';
-const buildQueryUrl = (assets: TTicker[], currencies: TTicker[]) => `
-  ${RATES_URL}/?fsyms=${assets.join(',')}&tsyms=${currencies.join(',')}
+const ASSET_ID_MAPPING_URL =
+  'https://raw.githubusercontent.com/MyCryptoHQ/assets/master/assets/assets.json';
+const ASSET_RATES_URL = 'https://api.coingecko.com/api/v3/simple/price';
+const buildAssetQueryUrl = (assets: string[], currencies: string[]) => `
+  ${ASSET_RATES_URL}/?ids=${assets}&vs_currencies=${currencies}
 `;
 
+const fetchAssetMappingList = async (): Promise<AssetMappingListObject | any> => {
+  return fetch(ASSET_ID_MAPPING_URL, {
```

Comment:
Prefer using the `ApiService` over `fetch`. It allows for a single place to deal with
- request headers
- failed calls
- response parsing
- throttling and retries.